python/dgl/mock_sparse/sddmm.py

```diff
@@ -0,0 +1,73 @@
+"""dgl sddmm operators for sparse matrix module."""
+import torch
+from .sp_matrix import SparseMatrix
+
+__all__ = ['sddmm']
+
+def sddmm(A: SparseMatrix, matB: torch.tensor, matC: torch.tensor) -> SparseMatrix:
+    r""" Sampled-Dense-Dense Matrix Multiplication.
```

Comment:
Please describe the operator behavior when `A` has vector-shaped non-zero values.

Reply:
If there is no concrete need for vector-shaped non-zero case, we should drop it for the MVP.

Reply:
We need for GAT example.

Reply:
I'm not 100% sure the behavior you described here is what GAT wants but let's leave it to the GAT PR.